src/baked_file_system.cr

```diff
@@ -11,28 +12,44 @@ module BakedFileSystem
     getter! mime_type : String
     getter! size : Int32
     getter! encoded : String
+    getter! compressed_size : Int32
 
     @slice   : Slice(UInt8)?
-    @string  : String?
+    @io  : IO?
 
-    def initialize(@path, @mime_type, @size, @encoded)
+    def initialize(@path, @mime_type, @size, @compressed_size, @encoded)
       @name = File.basename(path)
     end
 
-    def read
```

Comment:
Because it doesn't work for binary files.  `read` returns a String which must be UTF8.  Calling it on a PNG isn't safe afaik.